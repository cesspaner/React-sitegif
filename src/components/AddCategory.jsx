import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    
    const [inputValue, setinputValue] = useState('')
    
    const onInputChange = ({target})=>{
        // console.log(target.value)
        setinputValue(target.value)
    }
    
    const onSubmit =(event)=>{
        event.preventDefault()
        // console.log(inputValue)
        if (inputValue.trim().length <= 1) return

        // setCategories(categories =>[inputValue, ...categories])
        onNewCategory(inputValue.trim())
        setinputValue('')
    }
  return (
    <form  onSubmit={onSubmit}>
      <input
      type="Text"
      placeholder = "Buscar Gif"
      value={inputValue}
      onChange = {onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes ={
  onNewCategory: PropTypes.func.isRequired,
}