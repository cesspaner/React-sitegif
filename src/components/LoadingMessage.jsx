export const LoadingMessage = ({isLoading}) => {
  return (
        isLoading && (<h2 className="loading">Cargando...</h2>)//and logico
        // isLoading ? (<h2>Cargando...</h2>): null //ternario donde regreso el loading si es false
  )
}
