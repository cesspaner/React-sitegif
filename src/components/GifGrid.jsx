import { useFethGifs } from "../hooks/useFethGifs"
import { GifItem } from "./GifItem"
import { LoadingMessage } from "./LoadingMessage"

export const GifGrid = ({category}) => {

const {images, isLoading} = useFethGifs(category)


  return (
    <>
      <h3>{category}</h3>
      <LoadingMessage isLoading= {isLoading}/>
      <div className="card-grid">
        {
            images.map((image) =>(
                <GifItem 
                    key  ={image.id}
                    // title={image.title}
                    // url  ={image.url} 
                    {...image} //para mandar todas las propiedades
                />
            ))
        }
      </div>
     </>
  )
}
