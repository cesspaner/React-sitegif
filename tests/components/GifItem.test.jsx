import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('AÃ±adir PropTypes', () => {
    const title = "Saitama"
    const url   = "https://one-punch.com/"
  test('Valores obligatorios', () => {
     const {container}=render(<GifItem title={title} url={url}/>)
     expect(container).toMatchSnapshot()

    })

  test('Debe de mostrar la imagen con el Url y ALT indicado', () => {
      render(<GifItem title={ title } url={ url } />)
      //screen.debug() 
      //   expect(screen.getByRole('img').src).toBe( url )
      //   expect(screen.getByRole('img').alt).toBe( title )

    const { src , alt }= screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
    })
})
