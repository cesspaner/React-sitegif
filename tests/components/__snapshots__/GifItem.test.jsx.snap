// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AÃ±adir PropTypes Valores obligatorios 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
